client/client_test.go

```diff
@@ -6942,110 +6942,153 @@ func testExportAttestations(t *testing.T, sb integration.Sandbox) {
+			require.Nil(t, attest2.Predicate)
+			subjects = []intoto.Subject{{
+				Name: "/attestation.json",
 				Digest: map[string]string{
-					"sha256": bases[i].Desc.Digest.Encoded(),
+					"sha256": successDigest.Encoded(),
 				},
-			},
+			}}
+			require.Equal(t, subjects, attest2.Subject)
 		}
-		require.Equal(t, subjects, attest.Subject)
 
-		var attest2 intoto.Statement
-		require.NoError(t, json.Unmarshal(att.LayersRaw[1], &attest2))
+		cdAddress := sb.ContainerdAddress()
+		if cdAddress == "" {
+			return
+		}
+		client, err := containerd.New(cdAddress)
+		require.NoError(t, err)
+		defer client.Close()
+		ctx := namespaces.WithNamespace(sb.Context(), "buildkit")
 
-		require.Equal(t, "https://in-toto.io/Statement/v0.1", attest2.Type)
-		require.Equal(t, "https://example.com/attestations2/v1.0", attest2.PredicateType)
-		require.Nil(t, attest2.Predicate)
-		subjects = []intoto.Subject{{
-			Name: "/attestation.json",
-			Digest: map[string]string{
-				"sha256": successDigest.Encoded(),
+		for _, target := range targets {
+			err = client.ImageService().Delete(ctx, target, images.SynchronousDelete())
+			require.NoError(t, err)
+		}
+		checkAllReleasable(t, c, sb, true)
+	})
+
+	t.Run("local", func(t *testing.T) {
+		dir := t.TempDir()
+		_, err = c.Build(sb.Context(), SolveOpt{
+			Exports: []ExportEntry{
+				{
+					Type:      ExporterLocal,
+					OutputDir: dir,
+				},
 			},
-		}}
-		require.Equal(t, subjects, attest2.Subject)
-	}
+		}, "", frontend, nil)
+		require.NoError(t, err)
 
-	cdAddress := sb.ContainerdAddress()
-	if cdAddress == "" {
-		return
-	}
-	client, err := containerd.New(cdAddress)
-	require.NoError(t, err)
-	defer client.Close()
-	ctx := namespaces.WithNamespace(sb.Context(), "buildkit")
+		for _, p := range ps {
+			var attest intoto.Statement
+			dt, err := os.ReadFile(path.Join(dir, strings.ReplaceAll(platforms.Format(p), "/", "_"), "attestation.json"))
+			require.NoError(t, err)
+			require.NoError(t, json.Unmarshal(dt, &attest))
 
-	for _, target := range targets {
-		err = client.ImageService().Delete(ctx, target, images.SynchronousDelete())
-		require.NoError(t, err)
-	}
-	checkAllReleasable(t, c, sb, true)
+			require.Equal(t, "https://in-toto.io/Statement/v0.1", attest.Type)
+			require.Equal(t, "https://example.com/attestations/v1.0", attest.PredicateType)
+			require.Equal(t, map[string]interface{}{"success": true}, attest.Predicate)
+			require.Equal(t, []intoto.Subject{}, attest.Subject)
+
+			var attest2 intoto.Statement
+			dt, err = os.ReadFile(path.Join(dir, strings.ReplaceAll(platforms.Format(p), "/", "_"), "attestation2.json"))
+			require.NoError(t, err)
+			require.NoError(t, json.Unmarshal(dt, &attest2))
+
+			require.Equal(t, "https://in-toto.io/Statement/v0.1", attest2.Type)
+			require.Equal(t, "https://example.com/attestations2/v1.0", attest2.PredicateType)
+			require.Nil(t, attest2.Predicate)
+			subjects := []intoto.Subject{{
+				Name: "/attestation.json",
```

Comment:
Not quite sure what is going on here. I think we discussed that subject should be all files (so "greeting"?). Hash is sha256 of content only, no metadata. Directories ignored.

Reply:
Have updated, so that the `Self` subject will create a subject for each file, with filename as `name` and `digest` with the digest of the file contents.

I think we should leave the `Raw` subject alone, just like with the `image` exporter.